### ISSUE TITLE
Change packaged git-flow to AVH version

### ISSUE THREAD

**Issue by rcdailey** [git-for-windows/git#416]
Recently, gitflow has been introduced into the installer for Git for Windows. However, it's referring to the old and abandoned [nvie/git-flow repository][1]. According to an [issue filed there][2], many users have observed it is abandoned, and there have been no recent commits for years.

Instead, an active and reliable fork of git-flow named [Gitflow AVH Edition][3] should be used. The package in question seems to be [here][4].

[1]: https://github.com/nvie/gitflow
[2]: https://github.com/nvie/gitflow/issues/285
[3]: https://github.com/petervanderdoes/gitflow-avh
[4]: https://github.com/Alexpux/MSYS2-packages/blob/master/git-flow/PKGBUILD

**Comment by linquize** [comment]
The old one lags behind for 3 years.

**Comment by rcdailey** [comment]
At the moment I am working on this.

@dscho: How do you test these PKGBUILD? Can you explain what steps to take? Thanks.

**Comment by rcdailey** [comment]
Current `PKGBUILD` has this weird logic in `pkgver()` function:

```
pkgver() {
  cd "${srcdir}/${_realname}"

  # Need to create sane tags to allow `git describe` to work
  for t in $(git tag -l | grep '^[0-9]')
  do
    git rev-parse --quiet --verify v$t >/dev/null ||
    git tag v$t $t^2
  done
  
  git describe --tags |
  sed -e 's/^v//' -e 'y/-/./'
}
```

Why does it loop through all of the tags and create a version with a `v` prefix tagged at the 2nd parent?

**Comment by dscho** [comment]
@rcdailey why does it not surprise me that you open another ticket instead of a Pull Request. *sigh*

**Comment by rcdailey** [comment]
@dscho I told you I'm working on this. I asked some questions, if you properly read everything, maybe you would be able to help me.

Also for once can you please stop bitching. There is nothing wrong with an issue to accompany any PRs. It provides a medium for discussion, which is an excellent for auditing purposes. As someone doing this for Microsoft now, I expected more professional behavior from you.

You're constantly rude to many people I see file issues on your github projects. It's really unacceptable. If you want help, you have to be nice to the people that will possibly invest their free time to help.

If you have nothing useful to say, I'd much rather you stay off this topic. Manage it as needed but don't comment on here with sarcasm or other topics that do not directly contribute to the project. Also I believe you're in violation of your own guidelines you setup not long ago. I doubt you'd appreciate other people talking to you that way.

Anyway I'm testing this now, I have some PKGBUILD modifications, I'll see if I can get it working.

**Comment by rimrul** [comment]
>>@rcdailey why does it not surprise me that you open yet another ticket instead of a Pull Request. sigh

>There is nothing wrong with an issue to accompany any PRs. It provides a medium for discussion, which is an excellent for auditing purposes.

>You're constantly rude

You really where a bit harsh. I don't think github lets you open an empty pull request (without code) so I would have probably done the same as @rcdailey. It can be closed, once it's solved and everything will be fine.

>@dscho: How do you test these PKGBUILD? Can you explain what steps to take? Thanks.

Well, you build them with `makepkg -s` or `makepkg-mingw -s`. You can install the outputfile with `pacman -U <package-file-name.tar.xz>`. To test that result you will probably have to build and in stall [an installer](https://github.com/git-for-windows/git/wiki/Making-an-installer) or [a portable release](https://github.com/git-for-windows/git/wiki/Making-a-portable-Git). Some more details about `pacman` can be found in the [wiki](https://github.com/git-for-windows/git/wiki/Package-management) or on the web. For example [here](https://wiki.archlinux.org/index.php/Creating_packages), [here](https://wiki.archlinux.org/index.php/PKGBUILD) and [here](https://www.archlinux.org/pacman/PKGBUILD.5.html).

>Current PKGBUILD has this weird logic in pkgver() function:
Why does it loop through all of the tags and create a version with a v prefix tagged at the 2nd parent?

I can't help you with that. Maybe you could test what happens if you comment these lines out.

**Comment by rcdailey** [comment]
@rimrul Thanks for the help. What I did so far was run `$ makepkg -sLf` inside of `/usr/src/MSYS2-packages/git-flow`, which allowed me to at least test the changes to `PKGBUILD`. So far it looks like it ran well. Now I need to do the integration test: Somehow build the git sdk without an installer to see if I can access the commands. I'll read through the links you've provided to see if i can figure this out. If not, I will try the installer as you suggest.

Thanks again.

**Comment by dscho** [comment]
Okay, I was a bit harsh. But y'all have to admit that the OP had a track record of making me finish things after dropping the ball. If that is changing now, I offer my apologies and my enthusiasm for the future.

**Comment by rcdailey** [comment]
@dscho Apology accepted.

Concerning the "track record" nonsense:

Let's be realistic here. It's an open source community. You aren't my boss, I'm not paid to help you. 

* All of the technologies in use in this project (language, toolchains, script languages, etc) I have LITTLE to NO experience with.
* I am forced to spend my time googling for help outside of this community on how to use YOUR tooling because I don't get any help from you. That's time I could have been spending reading some guided steps from you and actually *executing* on something material to put in a PR.
* I have my own job, family, and other obligations to worry about.

I don't take pleasure in giving up on things I offer to do, but sometimes it's outside of my control, or (as noted above) I am distracted by other meta issues. The fact that you have this narrow minded definition of contribution is also an issue. I actually care about this project enough to come help give information, file issues, etc. to help keep things on paper. If you prefer ONLY pull requests and DO NOT want issues filed, then simply disable reports or make it clear what you actually expect. Stop taking everything so personally.

And on a final note, there was 1 issue I gave up on. And I guess giving up on 1 issue is what you would call a "track record"?

I'm tired of arguing with you man. You're incredibly difficult to deal with because you take everything personally and you get rude. Is that the thanks I get for being involved (not to the extent of numerous PRs as you so desire, but at least I invest my time here in other productive ways).

Now if we can FINALLY move away from this, and hopefully going forward @dscho will be less rude about people spending their time here, maybe you could help me figure out how to test this? I am trying to build the portable installer per instructions linked earlier by @rimrul, but I get the following output.

```
$ pacman -Syu
:: Synchronizing package databases...
 git-for-windows is up to date                       0.0   B  0.00B/s 00:00 [-----------------------------------------]   0%
 mingw32 is up to date
 mingw64 is up to date
 msys is up to date
:: Starting full system upgrade...
warning: mintty: local (1322-2) is newer than git-for-windows (1~2.0.3-2)
resolving dependencies...
looking for conflicting packages...

Packages (113) bash-4.3.042-1  bash-completion-2.1-6  ca-certificates-20150426-1  coreutils-8.24-1  curl-7.44.0-1
               dos2unix-7.3-1  file-5.24-1  gawk-4.1.3-1  gcc-libs-4.9.2-5  gettext-0.19.6-1  git-2.5.2-1
               git-extra-1.1.74.1a9278b-1  info-6.0-1  less-481-1  libasprintf-0.19.6-1  libcurl-7.44.0-1  libgcrypt-1.6.4-1
               libgettextpo-0.19.6-1  libgpg-error-1.20-1  libidn-1.32-1  libintl-0.19.6-1  libnettle-3.1.1-1
               libopenssl-1.0.2.d-1  libpcre-8.37-1  libpcre16-8.37-1  libpcre32-8.37-1  libpcrecpp-8.37-1
               libpcreposix-8.37-1  libreadline-6.3.008-5  libssh2-1.6.0-1  libtasn1-4.7-1
               mingw-w64-x86_64-binutils-2.25.1-1  mingw-w64-x86_64-bzip2-1.0.6-4  mingw-w64-x86_64-c-ares-1.10.0-2
               mingw-w64-x86_64-ca-certificates-20150426-2  mingw-w64-x86_64-crt-git-5.0.0.4573.628fdbf-1
               mingw-w64-x86_64-curl-7.44.0-2  mingw-w64-x86_64-expat-2.1.0-6  mingw-w64-x86_64-gcc-5.2.0-3
               mingw-w64-x86_64-gcc-ada-5.2.0-3  mingw-w64-x86_64-gcc-fortran-5.2.0-3
               mingw-w64-x86_64-gcc-libgfortran-5.2.0-3  mingw-w64-x86_64-gcc-libs-5.2.0-3
               mingw-w64-x86_64-gcc-objc-5.2.0-3  mingw-w64-x86_64-gdb-7.10-1  mingw-w64-x86_64-gdbm-1.11-3
               mingw-w64-x86_64-gettext-0.19.6-1  mingw-w64-x86_64-git-2.5.3.1.6ee1c9b-1
               mingw-w64-x86_64-git-doc-html-2.5.3.1.6ee1c9b-1  mingw-w64-x86_64-gmp-6.0.0-3
               mingw-w64-x86_64-gnutls-3.4.5-1  mingw-w64-x86_64-headers-git-5.0.0.4574.ce1ad45-1
               mingw-w64-x86_64-isl-0.14.1-2  mingw-w64-x86_64-libffi-3.2.1-3  mingw-w64-x86_64-libiconv-1.14-5
               mingw-w64-x86_64-libidn-1.32-2  mingw-w64-x86_64-libmangle-git-5.0.0.4509.2e5a9a2-1
               mingw-w64-x86_64-libssh2-1.6.0-2  mingw-w64-x86_64-libsystre-1.0.1-2  mingw-w64-x86_64-libtasn1-4.7-1
               mingw-w64-x86_64-libtre-git-r122.c2f5d13-4  mingw-w64-x86_64-libwinpthread-git-5.0.0.4573.628fdbf-1
               mingw-w64-x86_64-make-4.1.2351.a80a8b8-1  mingw-w64-x86_64-mpc-1.0.3-2  mingw-w64-x86_64-mpfr-3.1.3.p0-2
               mingw-w64-x86_64-ncurses-6.0.20150627-2  mingw-w64-x86_64-nettle-3.1.1-2  mingw-w64-x86_64-openssl-1.0.2.d-1
               mingw-w64-x86_64-p11-kit-0.23.1-3  mingw-w64-x86_64-pcre-8.37-2  mingw-w64-x86_64-python2-2.7.10-4
               mingw-w64-x86_64-readline-6.3.008-1  mingw-w64-x86_64-rtmpdump-git-r499.a107cef-3
               mingw-w64-x86_64-tcl-8.6.4-2  mingw-w64-x86_64-termcap-1.3.1-2  mingw-w64-x86_64-tk-8.6.4-2
               mingw-w64-x86_64-tools-git-5.0.0.4573.628fdbf-1  mingw-w64-x86_64-windows-default-manifest-6.4-2
               mingw-w64-x86_64-wineditline-2.101-4  mingw-w64-x86_64-winpthreads-git-5.0.0.4573.628fdbf-1
               mingw-w64-x86_64-zlib-1.2.8-9  mpfr-3.1.3.p0-1  msys2-keyring-r9.397a52e-1
               msys2-runtime-2.2.1.33617.df1f0a9-1  ncurses-6.0.20150627-2  openssh-7.1p1-1  openssl-1.0.2.d-1
               pacman-4.2.1.6198.439d376-1  pacman-mirrors-20150722-2  pcre-8.37-1  perl-5.22.0-2  perl-Encode-Locale-1.04-1
               perl-Error-0.17024-1  perl-HTML-Parser-3.71-3  perl-IO-Socket-SSL-2.016-1  perl-IO-stringy-2.111-1
               perl-MIME-tools-5.506-1  perl-MailTools-2.14-1  perl-Module-Build-0.4212-1  perl-Net-HTTP-6.09-1
               perl-Net-SMTP-SSL-1.02-1  perl-Net-SSLeay-1.70-1  perl-TermReadKey-2.33-1  perl-Test-Pod-1.50-1
               perl-URI-1.68-1  perl-YAML-Syck-1.29-1  perl-libwww-6.13-1  python2-2.7.10-1  subversion-1.9.1-1
               tzcode-2015.e-1  vim-7.4.872-2  which-2.21-2  xmlto-0.0.26-2

Total Download Size:      3.67 MiB
Total Installed Size:  1015.41 MiB
Net Upgrade Size:        36.32 MiB

:: Proceed with installation? [Y/n] y
:: Retrieving packages ...
 mingw-w64-x86_64-ncurses-6.0.20150627-2-any      1794.9 KiB   584M/s 00:00 [#########################################] 100%
 gcc-libs-4.9.2-5-x86_64                           646.8 KiB   316M/s 00:00 [#########################################] 100%
 openssl-1.0.2.d-1-x86_64                         1321.1 KiB   430M/s 00:00 [#########################################] 100%
(113/113) checking keys in keyring                                          [#########################################] 100%
(113/113) checking package integrity                                        [#########################################] 100%
(113/113) loading package files                                             [#########################################] 100%
(113/113) checking for file conflicts                                       [#########################################] 100%
error: failed to commit transaction (conflicting files)
mingw-w64-x86_64-git: /mingw64/libexec/git-core/git-worktree.exe exists in filesystem
mingw-w64-x86_64-git: /mingw64/libexec/git-core/mergetools/winmerge exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.2.3.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.3.9.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.2.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.3.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.4.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.5.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.6.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.7.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.8.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.9.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.5.0.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.5.1.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.5.2.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.5.3.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/fmt-merge-msg-config.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/git-worktree.html exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/git-worktree.txt exists in filesystem
Errors occurred, no packages were upgraded.
```

**Comment by rcdailey** [comment]
Also I tried specifying `-f` to `pacman` but that option is not recognized apparently. Most of my googling on these issues takes me to the Archlinux community, which is probably not a lot of help since the pacman here is not 100% exact it seems.

**Comment by dscho** [comment]
> You aren't my boss, I'm not paid to help you.

You know, the same holds in the other direction...

**Comment by clh92** [comment]
你说什么鬼东西啊？

**Comment by rcdailey** [comment]
@dscho But you'll help me if you respect your community guidelines and actually *want* contributions. By all means feel free to ignore my questions, but most likely I'll give up on this too as a result. I'll patiently await your assistance.

**Comment by rimrul** [comment]
>你说什么鬼东西啊？

Accordording to google this means "What the hell you say ah?". @clh92,  you seem a little upset. Why? Also, could you please post in english? It simplifies international communication a lot, if everyone agrees to speak english.

>Also I tried specifying -f to pacman but that option is not recognized apparently. Most of my googling on these issues takes me to the Archlinux community, which is probably not a lot of help since the pacman here is not 100% exact it seems.

This `pacman` should be as close to Arch's `pacman` as possible. But even Arch's `pacman` manpage does not list an option `-f`. Maybe you where looking for `--force`? Otherwise I only know about a `-f` option for `makepkg`.

>mingw-w64-x86_64-git: /mingw64/libexec/git-core/git-worktree.exe exists in filesystem
mingw-w64-x86_64-git: /mingw64/libexec/git-core/mergetools/winmerge exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.2.3.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.3.9.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.2.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.3.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.4.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.5.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.6.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.7.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.8.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.4.9.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.5.0.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.5.1.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.5.2.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/RelNotes/2.5.3.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/fmt-merge-msg-config.txt exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/git-worktree.html exists in filesystem
mingw-w64-x86_64-git-doc-html: /mingw64/share/doc/git-doc/git-worktree.txt exists in filesystem

@rcdailey `pacman -Syu` fails to install `mingw-w64-x86_64-git` and `mingw-w64-x86_64-git-doc-html` because the following files exist: 

    /mingw64/libexec/git-core/git-worktree.exe
    /mingw64/libexec/git-core/mergetools/winmerge
    /mingw64/share/doc/git-doc/RelNotes/2.2.3.txt
    /mingw64/share/doc/git-doc/RelNotes/2.3.9.txt
    /mingw64/share/doc/git-doc/RelNotes/2.4.2.txt
    /mingw64/share/doc/git-doc/RelNotes/2.4.3.txt
    /mingw64/share/doc/git-doc/RelNotes/2.4.4.txt
    /mingw64/share/doc/git-doc/RelNotes/2.4.5.txt
    /mingw64/share/doc/git-doc/RelNotes/2.4.6.txt
    /mingw64/share/doc/git-doc/RelNotes/2.4.7.txt
    /mingw64/share/doc/git-doc/RelNotes/2.4.8.txt
    /mingw64/share/doc/git-doc/RelNotes/2.4.9.txt
    /mingw64/share/doc/git-doc/RelNotes/2.5.0.txt
    /mingw64/share/doc/git-doc/RelNotes/2.5.1.txt
    /mingw64/share/doc/git-doc/RelNotes/2.5.2.txt
    /mingw64/share/doc/git-doc/RelNotes/2.5.3.txt
    /mingw64/share/doc/git-doc/fmt-merge-msg-config.txt
    /mingw64/share/doc/git-doc/git-worktree.html
    /mingw64/share/doc/git-doc/git-worktree.txt

I'd sugest you move them to another location using `mv` (or Ctrl+X;Ctrl+V) and try again.

**Comment by rcdailey** [comment]
Removing them is something I could have tried by hand, however I assumed a deeper issue. I'm not sure why there are there in the first place, I haven't done anything exceptional beyond using tooling as documented on the wiki.

I'll remove them and let you know how it goes.

**Comment by rcdailey** [comment]
So I was able to get pacman to install everything with `pacman -Syu`. After that, I made sure I did a fetch/ff merge on all the repositories under `/usr/src`. I went into my package directory (`/usr/src/MSYS2-packages/git-flow`) and ran `makepkg -s` after starting the terminal from `msys2_shell.bat`. It appeared to build fine.

I then went to `/usr/src/build-extras` and ran `./portable/release.sh 1.2.3.4-test`. Portable installer built just fine, but I still see the original version of git-flow in there, not the new version.

Can someone fill me in on the steps I'm missing here? Very frustrating that this isn't more automated and easier to pick up on.

**Comment by rimrul** [comment]
>Can someone fill me in on the steps I'm missing here? Very frustrating that this isn't more automated and easier to pick up on.

Yes. You didn't install the new package. Try `pacman-U packagename-version.tar.xz` between the `makepkg` and the `cd /usr/src/build-extra`.

**Comment by rcdailey** [comment]
I thought (rather assumed, because it seemed only rational) the installation of packages intended to go into the installer were automatically installed. Does this have to be done for every single package that goes into the installer?

**Comment by rimrul** [comment]
>I thought (rather assumed, because it seemed only rational) the installation of packages intended to go into the installer were automatically installed. Does this have to be done for every single package that goes into the installer?

It does install the git-flow package, but the official version, not your locally compiled one.

**Comment by rcdailey** [comment]
So I ran the `pacman -U` command as you recommended, but after doing the install command again in `build-extra`, the logs say it is downgrading the git-flow package. So it seems I'm still missing a step, some way to tell it to use the local version of the package instead of grabbing it remotely again.

**Comment by rimrul** [comment]
>So I ran the pacman -U command as you recommended, but after doing the install command again in build-extra, the logs say it is downgrading the git-flow package.

It shouldn't do that. I don't know why it does that. Maybe we could try to force a package version somewhere around [make-file-list.sh#31](https://github.com/git-for-windows/build-extra/blob/4c5e4423b492009c9740910b199d0790997c4c7e/make-file-list.sh#L31), but that would be quite an ugly workaround at best. I think we'll have to ask @dscho about this. Do you know why it does that? Do you have any idea how to properly prevent this?

**Comment by dscho** [comment]
> I think we'll have to ask @dscho about this. Do you know why it does that?

I do not have enough information: there is no clue in this thread which version of `git-flow` was actually built. As per the [`PKGBUILD`](https://github.com/git-for-windows/MSYS2-packages/blob/315a7496f20e0e818fbfac8b07e94991f0f212d9/git-flow/PKGBUILD#L5) it should be 0.4.1.108.g15aab26-1. And this is indeed the current version. Let me paste [what I tried](http://whathaveyoutried.com/) to verify this:

```sh
$ pacman -Sy --needed git-flow
:: Synchronizing package databases...
 git-for-windows is up to date0   B  0.00B/s 00:00 [---------------------]   0%
 mingw32 is up to date
 mingw64 is up to date
 msys is up to date
warning: git-flow-0.4.1.108.g15aab26-1 is up to date -- skipping
 there is nothing to do
```

**Comment by rimrul** [comment]
>I do not have enough information

Ok, let me gather the Information that I can. @rcdailey is working on migrating the git-flow package from the current NVIE edition to the actively maintained AVH edition. the current version of that [seems to be 1.8.0](https://github.com/petervanderdoes/gitflow-avh/releases/tag/1.8.0). Since that version is still older than this ticket, I'll assume @rcdailey is trying to integrate that version into his modified `PKGBUILD`. He built his version with with `makepkg` and installed the locally built package with `pacman -U`. While making an installer (might be portable) `pacman` logged, that it downgraded the `git-flow` package. This supports the assumption of a package version over 0.4.1.108. This is not he behaviour I've experienced with locally built packages. The pacman command that does this downgrade seems to be `pacman -S --needed --noconfirm mingw-w64-$ARCH-connect git-flow unzip docx2txt mingw-w64-$ARCH-antiword mingw-w64-$ARCH-xpdf >&2`

I don't think it is intended that `pacman -S --needed --noconfirm` downgrades installed packages. This seems to make testing unnescesarily hard. 

@rcdailey, could you please confirm the steps I've outlined above and the git-flow version you are trying to use? While you are at it could you post the output of the `pacman -U` command we've discussed earlier and the `pacman -Sy --needed git-flow` command @dscho posted(please run them in that order without other `pacman` commands inbetween)?

@dscho, is this the information, you need to help @rcdailey back on track? Do you know what causes this downgrade/how to prevent it?

**Comment by rcdailey** [comment]
@rimrul Thanks again for your help. I'm sorry that you have to proxy my questions over to @dscho for me, it seems he's ignoring me for some reason.

Your description is pretty accurate. You were able to deduce that information just by following discussion and no version numbers were required to get the gist of the issue. However to answer the questions given:

The version is indeed 1.8.0 on the AVH edition repository. Below is the PKGBUILD that I'm using. Note I've had no help on setting this up really so I tried my best to figure it out on my own. If anything looks incorrect feel free to comment.

```
_realname=git-flow
pkgname=("${_realname}")
pkgver=1.8.0.r58.gd441201
pkgrel=1
pkgdesc="AVH Edition of the git extensions to provide high-level repository operations for Vincent Driessen's branching model"
arch=('i686' 'x86_64')
license=('BSD')
depends=('git' 'util-linux')
url="http://nvie.com/posts/a-successful-git-branching-model/"
source=("${_realname}"::"git+https://github.com/petervanderdoes/gitflow-avh.git#branch=develop")
sha1sums=('SKIP')

prepare () {
  cd "${srcdir}/${_realname}"

  # Make sure that gitflow-shFlags is handled properly
  test true = "$(git config core.symlinks)" || {
    git config core.symlinks true
    git reset --hard
  }
}

package() {
  cd "${srcdir}/${_realname}"

  make prefix="${pkgdir}/usr/" install
}
```

Results of `pacman -U`:

```
$ pacman -U git-flow-1.8.0.r58.gd441201-1-x86_64.pkg.tar.xz
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) git-flow-1.8.0.r58.gd441201-1

Total Installed Size:  0.21 MiB
Net Upgrade Size:      0.12 MiB

:: Proceed with installation? [Y/n] y
(1/1) checking keys in keyring                                              [#########################################] 100%
(1/1) checking package integrity                                            [#########################################] 100%
(1/1) loading package files                                                 [#########################################] 100%
(1/1) checking for file conflicts                                           [#########################################] 100%
(1/1) checking available disk space                                         [#########################################] 100%
(1/1) upgrading git-flow                                                    [#########################################] 100%
```

Result of `pacman -Sy --needed git-flow`:

```
$ pacman -Sy --needed git-flow
:: Synchronizing package databases...
 git-for-windows is up to date                       0.0   B  0.00B/s 00:00 [-----------------------------------------]   0%
 mingw32 is up to date
 mingw64 is up to date
 msys is up to date
warning: downgrading package git-flow (1.8.0.r58.gd441201-1 => 0.4.1.108.g15aab26-1)
resolving dependencies...
looking for conflicting packages...

Packages (1) git-flow-0.4.1.108.g15aab26-1

Total Installed Size:   0.09 MiB
Net Upgrade Size:      -0.12 MiB

:: Proceed with installation? [Y/n] y
(1/1) checking keys in keyring                                              [#########################################] 100%
(1/1) checking package integrity                                            [#########################################] 100%
(1/1) loading package files                                                 [#########################################] 100%
(1/1) checking for file conflicts                                           [#########################################] 100%
(1/1) checking available disk space                                         [#########################################] 100%
(1/1) downgrading git-flow                                                  [#########################################] 100%
```

Let me know if you need more information.

**Comment by dscho** [comment]
> @dscho, is this the information, you need to help @rcdailey back on track? Do you know what causes this downgrade/how to prevent it?

Yes. [This line](https://github.com/git-for-windows/build-extra/blob/master/make-file-list.sh#L31) needs to be edited.

**Comment by dscho** [comment]
> I'm sorry that you have to proxy my questions over to @dscho for me, it seems he's ignoring me for some reason.

@rcdailey if you tried to be annoying, you may well have succeeded.

:stuck_out_tongue_winking_eye: 

Seriously again, I would really appreciate it if the tone of the conversation here could be improved. I see @rimrul being very helpful and respectful. That sets a good example to follow.

**Comment by rcdailey** [comment]
How does that line need to be edited? And arguably more importantly, *why* does it need to be edited? That's extremely unintuitive.

**Comment by dscho** [comment]
> How does that line need to be edited? And arguably more importantly, why does it need to be edited?

This line tries to make sure that your `git-flow` Is up-to-date before you build an installer. "Up-to-date" as per MSys2's Pacman repositories, that is.

So yes, if you want to test any out-of-line version, you have to make sure that `git-flow` is *not* forced to the official version. You can do that by deleting the `git-flow` name from the line.

**Comment by rcdailey** [comment]
I have a pull request up. Most of the time spent here was a learning exercise for me, so thanks to everyone that helped me along.

I targeted the PR to the git-for-windows fork of the MSYS2-packages repository. If this is not correct, please let me know where it should go. Yours is a fork as well so I'm not sure how far upstream you want this to go.

Secondly, this issue *technically* should belong to the repository in which the change is needed, because I am not sure if github will know that #416 refers to an issue in an unrelated sibling repository unless the issue is within that repository. Issue reporting is disabled on that fork so I am not sure what your intention on issue structure is.

Tested & verified my change locally via portable installer.

**Comment by nalla** [comment]
@dscho I wonder why this ticket did not close automatically after 6611d8a got merged? Maybe because the merge commit 46bce1f is not inside the _Git for Windows_ "structure"?

**Comment by dscho** [comment]
@nalla hmm. It should have been closed automatically. https://github.com/Alexpux/MSYS2-packages/commit/6611d8aebae99b845b7fa0575b4bbb3e78b3757a is reachable from its active branch, after all. Maybe it is only because it is cross org.
